libhl-crypto/amcl/src/bls381/big.rs

```diff
@@ -0,0 +1,1055 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+use arch;
+use arch::Chunk;
+
+//#[cfg(D32)]
+use arch::DChunk;
+
+use bls381::dbig::DBIG;
+use rand::RAND;
+
+pub const MODBYTES:usize = 48;
+pub const BASEBITS:usize = 58;
+
+pub const NLEN: usize = (1+((8*MODBYTES-1)/BASEBITS));
+pub const DNLEN: usize = 2*NLEN;
+pub const BMASK: Chunk= ((1<<BASEBITS)-1);
+pub const HBITS: usize=(BASEBITS/2);
+pub const HMASK: Chunk= ((1<<HBITS)-1);
+pub const NEXCESS:isize = (1<<((arch::CHUNK)-BASEBITS-1));
+pub const BIGBITS:usize = (MODBYTES*8);
+
+
+#[derive(Copy)]
+pub struct BIG {
```

Comment:
There are a lot duplication of similar (same?) logic between different curves. Are we really need in this duplication?
As far as I remember, in original rust codebase there is no such duplication.

Reply:
@mikelodder7 For amcl v3 Nikita K made PR that allows to avoid copy/paste of curves code and easily consume it as a crate. See https://github.com/milagro-crypto/amcl/pull/60 There is an agreement that it will be accepted soon. After this we plan to publish it in crates.io.

Reply:
The key idea to avoid duplication is in https://github.com/milagro-crypto/amcl/pull/60/files#diff-860ff95753813540aca3b511fcc27a65

Reply:
Agreed. But that's not the point of this PR. This PR is purely to merge crypto-lib here. I'll let you know about the next one which is on standby as soon as this one is merged.